src/Docker/DockerCli.cc

```diff
@@ -0,0 +1,89 @@
+#include "DockerCli.hh"
+
+#include <algorithm>
+#include <boost/process.hpp>
+#include <cctype>
+#include <fmt/core.h>
+#include <iostream>
+#include <set>
+#include <spdlog/spdlog.h>
+
+namespace Docker
+{
+    namespace bp = boost::process;
+
+    namespace DockerCli
+    {
+        namespace
+        {
+            bool containerNameIsValid(const std::string &name)
+            {
+                std::set<char> specialCharactersAllowed{ '-', '_' };
+                for (const char &c : name)
```

Comment:
```suggestion
                for (auto c : name)
```

Iterating by copy is better here because `sizeof(char)` < `sizeof(const char &)`